module/HorizonArm.cpp

```diff
@@ -6,22 +6,28 @@
 
 #include "HorizonArm.h"
 
-// コンストラクタ
-HorizonArm::HorizonArm() {}
+bool HorizonArm::onKeep = true;
+
+// アームを水平にする処理を有効/無効化する
+void HorizonArm::setOnKeep(bool keepFg)
+{
+  onKeep = keepFg;
+}
 
 // アームの角位置を目標値に保つ
 void HorizonArm::keepArm()
 {
-  // アームの角度が目標値(水平)になるまで、アームを動かす
   Measurer measurer;
   Controller controller;
   Pid pid(1.0, 0.3, 0.001, HORIZONTAL_ARM_COUNT);
 
+  // タスクとして並列処理
   while(true) {
-    int currentCount = measurer.getArmMotorCount();
-    if(currentCount == HORIZONTAL_ARM_COUNT) break;
-
-    controller.setArmMotorPwm(pid.calculatePid(currentCount));
+    if(onKeep) {
+      // アームの角度が目標値(水平)になるように、アームを動かす
+      int currentCount = measurer.getArmMotorCount();
+      controller.setArmMotorPwm(pid.calculatePid(currentCount));
+    }
```

Comment:
現在のkeepArm関数は、タスクとして並列処理することを前提として書かれている。
そのため、タスク以外の場所で呼び出せない（無限ループになる）。
keepArm関数の再利用性を高めるために、タスクとして並列処理するかどうかに関係なく、純粋にアームを水平にするだけの関数にしてほしい。

```suggestion
    int currentCount = measurer.getArmMotorCount();
    if(currentCount == HORIZONTAL_ARM_COUNT) break;
    controller.setArmMotorPwm(pid.calculatePid(currentCount));
```

while文の前で、
`if(onKeep == false) return;`